packages/server/src/automations/loopUtils.ts

```diff
@@ -29,7 +29,12 @@ function replaceBindingsRecursive(
           innerValue,
           `steps.${loopStepNumber}`
         )
-      } else if (typeof innerValue === "object" && innerValue !== null) {
+      } else if (
+        innerValue &&
+        typeof innerValue === "object" &&
+        !Array.isArray(innerValue) &&
+        Object.keys(innerValue).length > 0
```

Comment:
If you're wanting to detect that something is an object, `_.isPlainObject` is a bit clearer.

Reply:
this can be re-written to
```
import { isPlainObject } from "lodash"
....
innerValue && isPlainObject(innerValue) && Object.keys(innerValue).length > 0
```

Reply:
Done, thanks!